wfe/wfe.go

```diff
@@ -1100,7 +1100,7 @@ func (wfe *WebFrontEndImpl) Registration(ctx context.Context, logEvent *requestE
 		return
 	}
 
-	if len(update.Agreement) > 0 && update.Agreement != wfe.SubscriberAgreementURL {
+	if len(update.Agreement) > 0 && update.Agreement != currReg.Agreement && update.Agreement != wfe.SubscriberAgreementURL {
```

Comment:
This check seems like it would make more sense as part of RA's "Is this a diff, or the same object back" check. But that would mean that both the RA and the WFE would have to know about the SubscriberAgreementURL. Given that complexity, I think it makes sense to keep the check in WFE, but it should have a comment explaining why it's here.